test/powershell/Modules/ThreadJob/ThreadJob.Tests.ps1

```diff
@@ -0,0 +1,345 @@
+            Write-Output $using:GVar1
+        }
+
+        $results = $job | Receive-Job -Wait
+        $results[0] | Should Be $Var1
+        $results[1] | Should Be $Var2
+        $results[2] | Should Be $Var3
+        $results[3] | Should Be 2
+        $results[4] | Should Be $Var4
+        $results[5] | Should Be $global:GVar1
+    }
+
+    It 'ThreadJob with ScriptBlock and Using variables and argument list' {
+
+        $Var1 = "Hello"
+        $Var2 = 52
+        $job = Start-ThreadJob -ScriptBlock {
+            param ([string] $param1)
+
+            "$using:Var1 $param1 $using:Var2"
+        } -ArgumentList "There"
+
+        $results = $job | Receive-Job -Wait
+        $results | Should Be "Hello There 52"
+    }
+
+    It 'ThreadJob with ScriptFile' {
+
+        $job = Start-ThreadJob -FilePath $scriptFilePath1
+        $results = $job | Receive-Job -Wait
+        $results.Count | Should be 10
+        $results[9] | Should be "Hello 9"
+    }
+
+    It 'ThreadJob with ScriptFile and Initialization script' {
+
+        $job = Start-ThreadJob -FilePath $scriptFilePath1 -Initialization { "Goodbye" }
+        $results = $job | Receive-Job -Wait
+        $results.Count | Should be 11
+        $results[0] | Should be "Goodbye"
+    }
+
+    It 'ThreadJob with ScriptFile and Argument list' {
+
+        $job = Start-ThreadJob -FilePath $scriptFilePath2 -ArgumentList @("Hello","Goodbye")
+        $results = $job | Receive-Job -Wait
+        $results[0] | Should be "Hello"
+        $results[1] | Should be "Goodbye"
+    }
+
+    It 'ThreadJob with ScriptFile and piped input' {
+
+        $job = "Hello","Goodbye" | Start-ThreadJob -FilePath $scriptFilePath3
+        $results = $job | Receive-Job -Wait
+        $results[0] | Should be "Hello"
+        $results[1] | Should be "Goodbye"
+    }
+
+    It 'ThreadJob with ScriptFile and Using variables' {
+
+        $Var1 = "Hello!"
+        $Array1 = 1..10
+
+        $job = Start-ThreadJob -FilePath $scriptFilePath4
+        $results = $job | Receive-Job -Wait
+        $results[0] | Should be $Var1
+        $results[1] | Should be 3
+        $results[2] | Should be $Array1
+    }
+
+    It 'ThreadJob with ScriptFile and Using variables with argument list' {
+
+        $Var1 = "There"
+        $Var2 = 60
+        $job = Start-ThreadJob -FilePath $scriptFilePath5 -ArgumentList "Hello"
+        $results = $job | Receive-Job -Wait
+        $results | Should Be "Hello There 60"
+    }
+
+    It 'ThreadJob with terminating error' {
+
+        $job = Start-ThreadJob -ScriptBlock { throw "MyError!" }
+        $job | Wait-Job
+        $job.JobStateInfo.Reason.Message | Should Be "MyError!"
+    }
+
+    It 'ThreadJob ThrottleLimit and Queue' {
+
+        try
+        {
+            # Start four thread jobs with ThrottleLimit set to two
+            Get-Job | where PSJobTypeName -eq "ThreadJob" | Remove-Job -Force
+            Start-ThreadJob -ScriptBlock { Start-Sleep -Seconds 60 } -ThrottleLimit 2
```

Comment:
This is not related to review; but for my education:
The ThrottleLimit syntax looks interesting; couple of questions:
* What is the scope of ThrottleLimit ? e.g. "all threadjobs in current script" ?
* How do subsequent Start-ThreadJob's in the script know under what ThrottleLimit they operate?

Reply:
Throttle limit is per PowerShell session so it limits all concurrent threadjobs created in that session.  The queue is static (singleton).  I currently don't have a way to get the current throttle limit value ... you just set it.  But something like that can be added.

See source code for more details:
https://github.com/PaulHigin/PSThreadJob/blob/master/ThreadJob/ThreadJob/PSThreadJob.cs#L18